Add active_liquidity module

cujowolf opened the pull request:
oif.models:
- Add new Balance dataclass for use with ERC20 token balances

oif.helpers:
- Add new function `get_erc20_token_balances` in alchemy helpers

oif.constants:
- Add new module to store constants such as token and allocator addresses for OIF

oif.active_liquidity:
- Add new function `get_adai_transfers`
- Add new function `get_adai_balances`

cujowolf committed:
Add active_liquidity module

oif.models:
- Add new Balance dataclass for use with ERC20 token balances

oif.helpers:
- Add new function `get_erc20_token_balances` in alchemy helpers

oif.constants:
- Add new module to store constants such as token and allocator addresses for OIF

oif.active_liquidity:
- Add new function `get_adai_transfers`
- Add new function `get_adai_balances`

cujowolf committed:
Remove local ethereum node

cvauclair commented on oif/active_liquidity.py:
```diff
@@ -0,0 +1,52 @@
+# AAVE Token Transfers
+# ================================================================
+
+def get_adai_transfers(allocator_addresses: list[str], start_block: int, end_block: int):
```
@cujowolf Id suggest re-writing this function using the `alchemy_getAssetTransfers` JSON RPC method. This is a method that Alchemy offers as part of its "extended web3 API" (see https://docs.alchemy.com/alchemy/enhanced-apis/transfers-api).

Here's a code snippet to illustrate how it works:
```python
ALCHEMY_URL = f'https://eth-mainnet.alchemyapi.io/v2/{ALCHEMY_API_KEY}'

def get_erc20_transfers_from(wallet_addr: str, token_addr: str | list[str], startblock: int = 0, endblock: int = 99999999):
  req = {
    "jsonrpc": "2.0",
    "id": 0,
    "method": "alchemy_getAssetTransfers",
    "params": [
      {
        "fromBlock": str(hex(startblock)),
        "toBlock": str(hex(endblock)),
        "fromAddress": wallet_addr,
        "contractAddresses": [token_addr] if type(token_addr) == str else token_addr,
        "excludeZeroValue": True,
        "category": ["erc20"]
      }
    ]
  }

  resp = requests.post(ALCHEMY_URL, json=req).json()
  return resp['result']['transfers']

def get_erc20_transfers_to(wallet_addr: str, token_addr: str | list[str], startblock: int = 0, endblock: int = 99999999):
  req = {
    "jsonrpc": "2.0",
    "id": 0,
    "method": "alchemy_getAssetTransfers",
    "params": [
      {
        "fromBlock": str(hex(startblock)),
        "toBlock": str(hex(endblock)),
        "toAddress": wallet_addr,
        "contractAddresses": [token_addr] if type(token_addr) == str else token_addr,
        "excludeZeroValue": True,
        "category": ["erc20"]
      }
    ]
  }

  resp = requests.post(ALCHEMY_URL, json=req).json()
  return resp['result']['transfers']

def get_erc20_transfers(
  wallet_addr: str,
  token_addr: str | list[str],
  startblock: int = 0,
  endblock: int = 99999999,
  get_price: Callable = None
):
  transfers = (
    get_erc20_transfers_from(wallet_addr, token_addr, startblock, endblock) +
    get_erc20_transfers_to(wallet_addr, token_addr, startblock, endblock)
  )

  transfers.sort(key=lambda transfer: int(transfer['blockNum'], 16))

  return pd.DataFrame(transfers)
```

cujowolf replied:
This is updated and the performance increase over the previous functions is drastically improved.

cvauclair reviewed (changes requested):
See comment on `oif/active_liquidity.py` for `get_adai_transfers` function

cujowolf committed:
Use extended Alchemy API for get_adai_transfers

oif.active_liquidity:
- Add helpers for the Alchemy Extended API calls
- modify `get_dai_transfers` to utilize these helpers for faster processing
